src/components/ChangesNotAppliedModal.jsx

```diff
@@ -1,37 +1,51 @@
 import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
 import {
   Modal, Typography, Space, Button,
 } from 'antd';
+import { useRouter } from 'next/router';
+import { discardChangedQCFilters, navigateFromProcessingTo } from '../redux/actions/experimentSettings';
+import { runPipeline } from '../redux/actions/pipeline';
 
 import { getUserFriendlyQCStepName } from '../utils/qcSteps';
 
 const { Text } = Typography;
 
 const ChangesNotAppliedModal = (props) => {
-  const {
-    steps, visible, onRun, onDiscard, onCancel,
-  } = props;
+  const { experimentId } = props;
+
+  const router = useRouter();
+  const dispatch = useDispatch();
+
+  const { changedQCFilters, navigationPath } = useSelector((state) => state.experimentSettings.processing.meta);
```

Comment:
Why do we store navigationPath in redux?

Reply:
Because it has to be shared between multiple components on different hierarchy levels. The link in ContentWrapper needs it, and the "finish QC" button inside the Data Processing component also needs to know about it.